websites/permissions.py

```diff
@@ -0,0 +1,245 @@
+""" permissions for websites """
+from django.contrib.auth.models import Group, Permission
+from django.db import transaction
+from guardian.shortcuts import assign_perm, get_perms
+from rest_framework.generics import get_object_or_404
+from rest_framework.permissions import BasePermission, SAFE_METHODS
+
+from users.models import User
+from websites import constants
+from websites.models import Website
+
+
+def assign_website_permissions(group_or_user, perms, website=None):
+    """
+    Assign appropriate permissions to a website admin group
+
+    Args:
+        group_or_user (Group or User): the group/user to add permissions to
+        website (Object): The object that the permissions are for (typically Website)
+        perms (list of str): The permissions to be assigned
+
+    Returns:
+        bool: True if any permissions were added, False otherwise
+    """
+    if website:
+        current_perms = set(get_perms(group_or_user, website))
+    elif isinstance(group_or_user, Group):
+        current_perms = {perm.codename for perm in group_or_user.permissions.all()}
+    elif isinstance(group_or_user, User):
+        current_perms = {
+            perm.codename for perm in Permission.objects.filter(user=group_or_user)
+        }
+    else:
+        raise TypeError("Permissions must be assigned to a user or group")
+    if not {perm.replace("websites.", "") for perm in perms}.difference(current_perms):
+        return False
```

Comment:
This will help a little with readability I think

```suggestion
    if website:
        current_perm_codenames = set(get_perms(group_or_user, website))
    elif isinstance(group_or_user, Group):
        current_perm_codenames = {perm.codename for perm in group_or_user.permissions.all()}
    elif isinstance(group_or_user, User):
        current_perm_codenames = {
            perm.codename for perm in Permission.objects.filter(user=group_or_user)
        }
    else:
        raise TypeError("Permissions must be assigned to a user or group")
    added_perm_codenames = {perm.replace("websites.", "") for perm in perms}
    if not added_perm_codenames.difference(current_perm_codenames):
        return False
```